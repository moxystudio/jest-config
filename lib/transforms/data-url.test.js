'use strict';

const fs = require('fs');
const path = require('path');
const tmp = require('tmp-sync');
const dataUrlTransformer = require('./data-url');

const dir = tmp.in(__dirname);

afterAll(() => {
    tmp.clean();
});

it('should return string containing data-url', () => {
    const contents = 'this is a standard text file!';
    const filename = `${dir}/dataurl-test-file.txt`;

    // Can be verified at https://regex101.com/r/r8tE5u/3
    const matcher = /data:([\w/+]+);(charset=[\w-]+|base64),([a-zA-Z0-9+/]+={0,2})/;

    fs.writeFileSync(filename, contents);

    const output = dataUrlTransformer.process(contents, filename, {});

    expect(output).toMatch(matcher);
    expect(output).toMatchSnapshot();
});

it('should throw when it cannot guess the MIME type of a file and show extension', () => {
    const contents = 'this is a test file with a strange extension!';
    const filename = `${dir}/dataurl-test-file.strange`;

    fs.writeFileSync(filename, contents);

    expect(() => dataUrlTransformer.process(contents, filename, {})).toThrow(/Could not guess/);
    expect(() => dataUrlTransformer.process(contents, filename, {})).toThrow(path.extname(filename));
});

it('should throw when it cannot guess the MIME type of a file and show whole filename when there is no extension', () => {
    const contents = 'this is a test file with a strange extension!';
    const filename = `${dir}/dataurl-test-file-strange`;

    fs.writeFileSync(filename, contents);

    expect(() => dataUrlTransformer.process(contents, filename, {})).toThrow(/Could not guess/);
    expect(() => dataUrlTransformer.process(contents, filename, {})).toThrow(path.basename(filename));
});
