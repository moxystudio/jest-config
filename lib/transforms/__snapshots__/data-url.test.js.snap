// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return string containing data-url 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;
const dataUrl = \\"data:text/plain;base64,dGhpcyBpcyBhIHN0YW5kYXJkIHRleHQgZmlsZSE=\\";
var _default = dataUrl;
exports.default = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImRhdGF1cmwtdGVzdC1maWxlLnR4dCJdLCJuYW1lcyI6WyJkYXRhVXJsIl0sIm1hcHBpbmdzIjoiOzs7Ozs7QUFDQSxNQUFNQSxPQUFPLEdBQUcsaUVBQWhCO2VBRWVBLE8iLCJzb3VyY2VzQ29udGVudCI6WyJcbmNvbnN0IGRhdGFVcmwgPSBcImRhdGE6dGV4dC9wbGFpbjtiYXNlNjQsZEdocGN5QnBjeUJoSUhOMFlXNWtZWEprSUhSbGVIUWdabWxzWlNFPVwiO1xuXG5leHBvcnQgZGVmYXVsdCBkYXRhVXJsO1xuIl19"
`;
